master/internal/resourcemanagers/scheduler_config.go

```diff
@@ -0,0 +1,87 @@
+package resourcemanagers
+
+import (
+	"github.com/determined-ai/determined/master/pkg/check"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/union"
+)
+
+const (
+	defaultSchedulingPriority = 42
+
+	fairShareScheduling = "fair_share"
+	priorityScheduling  = "priority"
+
+	best             = "best"
+	worst            = "worst"
+	defaultFitPolicy = best
+)
+
+func defaultSchedulerConfig() *SchedulerConfig {
+	return &SchedulerConfig{
+		FairShare:     &FairShareSchedulerConfig{},
+		FittingPolicy: defaultFitPolicy,
+	}
+}
+
+// SchedulerConfig holds the configurations for scheduling policies.
+type SchedulerConfig struct {
+	FairShare     *FairShareSchedulerConfig `union:"type,fair_share" json:"-"`
+	Priority      *PrioritySchedulerConfig  `union:"type,priority" json:"-"`
+	FittingPolicy string                    `json:"fitting_policy"`
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+func (s SchedulerConfig) MarshalJSON() ([]byte, error) {
+	return union.Marshal(s)
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (s *SchedulerConfig) UnmarshalJSON(data []byte) error {
+	return union.Unmarshal(data, s)
+}
+
+// Validate implements the check.Validatable interface.
+func (s SchedulerConfig) Validate() []error {
+	return []error{
+		check.Contains(
+			s.FittingPolicy, []interface{}{"best", "worst"}, "invalid fitting policy",
```

Comment:
blocking: here should be consistent where you use the constant `best` and `worst` although I personally prefer directly using the strings though.